### ISSUE TITLE
Missing SSL Configuration reference for MySQL/MariaDB

### ISSUE THREAD

**Issue by derrickmehaffy** [knex/knex#5933]
In the new documentation there is no reference on how to configure SSL (either to disable it, reject unauthorized, or provide the cert, key, and ca files)

**Comment by mercmobily** [comment]
Hello, thanks for pointing this out.

This is a documentation gap; SSL/TLS options for MySQL/MariaDB are supported by passing the driver’s `ssl` configuration through Knex, but the new docs lack a clear example.

Knex forwards the `connection` object to the MySQL driver as-is when creating connections (lib/dialects/mysql/index.js:65-79). To enable TLS you can use the driver’s `ssl` option. For `mysql2`:

```
// knexfile.js
const fs = require('fs');

module.exports = {
  client: 'mysql2',
  connection: {
    host: process.env.DB_HOST,
    user: process.env.DB_USER,
    password: process.env.DB_PASS,
    database: process.env.DB_NAME,
    ssl: {
      rejectUnauthorized: true,
      ca: fs.readFileSync('ca.pem', 'utf8'),
      cert: fs.readFileSync('client-cert.pem', 'utf8'),
      key: fs.readFileSync('client-key.pem', 'utf8'),
      minVersion: 'TLSv1.2',
    },
  },
};
```

Variations:

- To disable verification in development: `ssl: { rejectUnauthorized: false }` or `ssl: 'Amazon RDS'` for certain managed services.
- For Sockets/Cloud SQL, combine SSL with `socketPath` as needed.

In conclusion, SSL is supported via the underlying driver options and simply passed through by Knex. We should add a dedicated docs section with examples like the above for MySQL/MariaDB.
